contracts/core/ModuleCore.sol

```diff
@@ -19,7 +19,7 @@ import {ContextUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/Cont
  * @author Cork Team
  * @notice Modulecore contract for integrating abstract modules like PSM and Vault contracts
  */
-contract ModuleCore is OwnableUpgradeable, UUPSUpgradeable, PsmCore, Initialize, VaultCore {
+abstract contract ModuleCore is OwnableUpgradeable, UUPSUpgradeable, PsmCore, Initialize, VaultCore {
```

Comment:
This shouldn't be abstract, all changes to function signature must also be followed with interface adjustments. All changes related to PSM functionality, should be done in `Psm.sol`. Same for vault must be done in  `Vault.sol`